dataclients/kubernetes/definitions/routegroupvalidator.go

```diff
@@ -56,7 +64,10 @@ func (rgv *RouteGroupValidator) filtersValidation(item *RouteGroupItem) error {
 	var errs []error
 	for _, r := range item.Spec.Routes {
 		for _, f := range r.Filters {
-			_, err := eskip.ParseFilters(f)
+			filters, err := eskip.ParseFilters(f)
+			if len(filters) != 1 && err == nil {
```

Comment:
but I want both, I don't want one or either. If we have 2 filters and one of them failed parsing then I want 2 errors to appear in validation one for parsing and one for number of filters

Reply:
>  If we have 2 filters and one of them failed parsing

In such case I think eskip.ParseFilters(f) will just return an error (you can check).
It is an established go pattern to check error first and if it is not nil ignore the result (which is often empty or malformed so cant be relied on anyways).

Reply:
> In such case I think eskip.ParseFilters(f) will just return an error (you can check).

No, it won't.

```
rzavodskikh@ZALANDO-39125:/tmp/tmp$ cat main.go 
package main

import (
	"fmt"

	"github.com/zalando/skipper/eskip"
)

func main() {
	filters, err := eskip.ParseFilters("status(201) -> inlineContent(\"hi\")")
	fmt.Printf("filters: %+v, err: %+v\n", filters, err)
}
rzavodskikh@ZALANDO-39125:/tmp/tmp$ go build main.go
rzavodskikh@ZALANDO-39125:/tmp/tmp$ ./main 
filters: [status(201) inlineContent("hi")], err: <nil>
rzavodskikh@ZALANDO-39125:/tmp/tmp$ 

```

Reply:
> In such case I think eskip.ParseFilters(f) will just return an error (you can check).
Yes, it will. Moreover, filters variable would be empty in this case, even if first filter is correct. So, I do agree it's better not to return two error in this case, one would be enough.

```
rzavodskikh@ZALANDO-39125:/tmp/tmp$ cat main.go 
package main

import (
	"fmt"

	"github.com/zalando/skipper/eskip"
)

func main() {
	filters, err := eskip.ParseFilters("status(201) -> inlineContent(\"hi\"")
	fmt.Printf("filters: %+v, err: %+v\n", filters, err)
}
rzavodskikh@ZALANDO-39125:/tmp/tmp$ go build main.go
rzavodskikh@ZALANDO-39125:/tmp/tmp$ ./main 
filters: [], err: parse failed after token ->, last route id: , position 41: syntax error
rzavodskikh@ZALANDO-39125:/tmp/tmp$ 
```

Reply:
I don't get the problem between the comment and what I wrote.
`ParseFilters` either return an error or list of filters

In case of list of filters I want to make sure there is only one filter, and in case of error I want to return the error with other errors. I don't want to only return this specific error because it will cause more roundtrips of fixes.

Reply:
Discussed this part of code on a call, I showed one more time that ParseFilters may return:
+ non-empty filters slice XOR
+ not nil error
but it never returns them at the same call.

So, from my pov it does not make sense try to put two errors at the `errs` slice here, so the code of Alexander at the beginning from this discussion makes the most sense to me here.